aten/src/ATen/core/type.cpp

```diff
@@ -1098,7 +1098,7 @@ bool TensorType::isSubtypeOfExt(const TypePtr& rhs, std::ostream* why_not) const
     if (this == rhs_p.get()) {
       return true;
     }
-    return *merge(*rhs_p) == *rhs_p;
```

Comment:
cc @Krovatkin for review. the logic here was wrong previously. If you have:
RHS: Tensor(15, 5)
This: Tensor
Merge: Tensor

Then this subtypes RHS because the merge is equal to current type. we were checking if rhs subtypes this previously